pkg/apiserver/logsearch/router.go

```diff
@@ -0,0 +1,357 @@
+	endpoint.POST("/tasks/run/:id", s.TaskRun)
+	endpoint.GET("/tasks/download/:id", s.TaskDownload)
+	endpoint.GET("/tasks/download", s.MultipleTaskDownload)
+	endpoint.POST("/tasks/cancel/:id", s.TaskCancel)
+	endpoint.DELETE("/tasks/:id", s.TaskDelete)
+}
+
+// @Summary List tasks
+// @Description list all log search tasks
+// @Produce json
+// @Success 200 {array} TaskModel
+// @Failure 400 {object} httputil.HTTPError
+// @Router /tasks [get]
+func (s *Service) TaskGetList(c *gin.Context) {
+	var tasks []TaskModel
+	err := s.db.Find(&tasks).Error
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	c.JSON(http.StatusOK, tasks)
+}
+
+// @Summary Preview logs by task ID
+// @Description preview fetched logs in a task
+// @Produce json
+// @Param id path string true "task id"
+// @Success 200 {array} PreviewModel
+// @Failure 400 {object} httputil.HTTPError
+// @Router /tasks/preview/{id} [get]
+func (s *Service) TaskPreview(c *gin.Context) {
+	taskID := c.Param("id")
+	var lines []PreviewModel
+	err := s.db.Where("task_id = ?", taskID).Find(&lines).Error
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	c.JSON(http.StatusOK, lines)
+}
+
+// @Summary Preview logs by task IDs
+// @Description preview fetched logs in multiple tasks
+// @Produce json
+// @Param id query string true "task id"
+// @Success 200 {array} LinePreview
+// @Failure 400 {object} httputil.HTTPError
+// @Router /tasks/preview/ [get]
+func (s *Service) MultipleTaskPreview(c *gin.Context) {
+	ids := c.QueryArray("id")
+	previews, err := s.getPreviews(ids)
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	res := mergeLines(previews)
+	c.JSON(http.StatusOK, res)
+}
+
+func (s *Service) getPreviews(ids []string) ([]*LogPreview, error) {
+	previews := make([]*LogPreview, 0, len(ids))
+	for _, taskID := range ids {
+		task, err := s.queryTaskByID(taskID)
+		if err != nil {
+			return nil, err
+		}
+		var lines []PreviewModel
+		err = s.db.Where("task_id = ?", taskID).Find(&previews).Error
+		if err != nil {
+			return nil, err
+		}
+		previews = append(previews, &LogPreview{
+			task,
+			lines,
+		})
+	}
+	return previews, nil
+}
+
+// @Summary Create task group
+// @Description create and run a task group
+// @Produce json
+// @Failure 400 {object} httputil.HTTPError
+// @Failure 500 {object} httputil.HTTPError
+// @Router /tasks [post]
+func (s *Service) TaskGroupCreate(c *gin.Context) {
+	// TODO: using parameters provided by client
+	endTime := time.Now().UnixNano() / int64(time.Millisecond)
+	startTime := int64(0)
+	var searchLogReq = &SearchLogRequest{
+		StartTime: startTime,
+		EndTime:   endTime,
+		Levels:    nil,
+		Patterns:  nil,
+	}
+	var components = []*Component{
+		{
+			ServerType: "tidb",
+			IP:         "127.0.0.1",
+			Port:       "4000",
+			StatusPort: "10080",
+		},
+	}
+	taskGroupID := scheduler.addTasks(components, searchLogReq)
```

Comment:
Does a single search request contain multiple tasks?

Reply:
Yes, a request will start a `taskGroup`, and it will start multiple tasks (one component per task).